src/app/pages/deposits/deposits.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Deposit } from 'src/app/models/Deposit';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ToastrService } from 'ngx-toastr';
+import { NgxSpinnerService } from 'ngx-spinner';
+import { Subscription } from 'rxjs';
+
+import { DepositsService } from 'src/app/services/deposits/deposits.service';
+import { removeSubscription } from 'src/app/helpers/unsubscribe';
+
+@Component({
+  selector: 'app-deposits',
+  templateUrl: './deposits.component.html',
+  styleUrls: ['./deposits.component.scss']
+})
+export class DepositsComponent implements OnInit, OnDestroy {
+  subscribe: Subscription[] = [];
+  headElements = ['Reference', 'Date', 'Amount'];
+  transactions: Deposit[] = [];
+  startDateVal: string = '';
+  endDateVal: string = '';
+  newResults: any[];
+
+  constructor(
+    private depositsService: DepositsService,
+    private spinner: NgxSpinnerService,
+    private toastrService: ToastrService
+  ) {}
+
+  onFilter() {
+    const filteredTransactions = [];
+    this.transactions.filter(transaction => {
+      const date = new Date(transaction.created_at);
```

Comment:
Hey @Genza999 , kindly add the `created_at` date to the `Deposits model` file in the `transaction definition`

```
  transaction?: {
    balance: string;
    status: string;
    created_at: string; // this should also be here
  };
```

Reply:
Hey @Tevinthuku , it actually doesn't have to be there but I have decided to change the name used on line 31 to deposit and not transaction as I think it probably was confusing.

Reply:
Nice, I guess we are reaping the benefits of `type-safety`